lib/applications/resolvers.ts

```diff
@@ -386,6 +395,9 @@ export const createNewApplication: Resolver<
       data: {
         type: 'NEW',
         donationAmount: donationAmount || 0,
+        paymentMethod2: hasSecondPaymentMethod ? paymentMethod2 : null,
+        processingFee2: hasSecondPaymentMethod ? processingFee2 || 0 : null,
+        donationAmount2: hasSecondPaymentMethod ? donationAmount2 || 0 : null,
```

Comment:
Why is `hasSecondPaymentMethod` necessary? Do `paymentMethod2`, `processingFee2`, and `donationAmount2` default to null?

Reply:
`hasSecondPaymentMethod` helps differentiate the following types of applications since `paymentMethod2` is null in both cases:

1. An application without a second payment method
2. An application where a user has selected that they would like to add a second payment method but has yet to choose which method

Although there is an approach that does not require `hasSecondPaymentMethod`, this solution significantly reduces complexity in `validation.ts` and maintains a similar schema design as the other application boolean properties such as `shippingAddressSameAsHomeAddress` and `billingAddressSameAsHomeAddress` that are linked to the setting of the shipping and billing address, respectively.